#!/usr/bin/python

## File launcher
#
# Freemind problem: "file:" scheme not recognized, therefore .mm files not opening.
# Proxy calls through "freemind.starter".
#

import os, sys

def help(quit=False):
    print 'Usage:\t',
    print '\033[34mrun <program> {<argument>}\033[0m'
    print 'This runs \033[34m<program>\033[0m detached from a terminal emulator.'
    if quit:
        sys.exit(0)

argv,debug = sys.argv[1:],False
if not argv:
    help(True)
if argv[0] in ('-d','--debug'):
    argv,debug = argv[1:],True
if debug or os.fork()==0:
    cmd = ' '.join('"%s"' %s for s in argv)
    redir = '' if debug else ' >/dev/null 2>&1'
    os.system('(%s || mimeopen "$(realpath %s)")%s' %(cmd,cmd,redir))
